src/domain/session/SessionViewModel.js

```diff
@@ -112,9 +118,15 @@ export class SessionViewModel extends ViewModel {
         return this._roomViewModelObservable?.get();
     }
 
+    get roomInfoViewModel() {
+        return this._roomInfoViewModel;
+    }
+
     _updateGrid(roomIds) {
         const changed = !(this._gridViewModel && roomIds);
         const currentRoomId = this.navigation.path.get("room");
+        // Close right-panel if needed
+        this._toggleRoomInformationPanel();
```

Comment:
you should not have to react to a change in the grid rooms set, as the active room will be a room segment after that. So you can just react to a change in the active room, and it will work for both grid and single rooms.

So, concretely, you can call _toggleRoomInformationPanel from `this.track(currentRoomId.subscribe(roomId => ` on line 62, and in your toggle method check if the room id of the current vm, if any is still the active one.